pkg/endpoints/dashboard.go

```diff
@@ -130,72 +106,173 @@ func IsOpenShift(r Resource, installedNamespace string) bool {
-			if strings.Contains(deploymentImage, "pipeline/cmd/controller") && strings.Contains(deploymentImage, ":") && strings.Contains(deploymentImage, "@") {
-				s := strings.SplitAfter(deploymentImage, ":")
-				if strings.Contains(s[1], "@") {
-					t := strings.Split(s[1], "@")
-					version = t[0]
+			if namespace == "openshift-pipelines" {
+				deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+				if strings.Contains(deploymentImage, "openshift-pipeline/tektoncd-"+thingSearchingFor+"-controller") && strings.Contains(deploymentImage, ":") {
+					s := strings.SplitAfter(deploymentImage, ":")
+					if s[1] != "" {
+						version = s[1]
+					}
+				}
+			}
+		}
+	}
+
+	if version == "" && thingSearchingFor == "pipelines" {
+		for _, deployment := range deployments.Items {
+			deploymentAnnotations := deployment.Spec.Template.GetAnnotations()
+
+			if namespace == "openshift-pipelines" {
+				deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+				if strings.Contains(deploymentImage, "openshift-pipeline/tektoncd-pipeline-controller") && strings.Contains(deploymentImage, ":") {
+					s := strings.SplitAfter(deploymentImage, ":")
+					if s[1] != "" {
+						version = s[1]
+					}
+				}
+			}
+
+			// For master of Tekton Pipelines
+			if version == "" {
+				version = deploymentAnnotations["pipeline.tekton.dev/release"]
+			}
+
+			// For Tekton Pipelines 0.10.0 + 0.10.1
+			if version == "" {
+				version = deploymentAnnotations["tekton.dev/release"]
+			}
+
+			// For Tekton Pipelines 0.9.0 - 0.9.2
+			if version == "" {
+				deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+				if strings.Contains(deploymentImage, "pipeline/cmd/controller") && strings.Contains(deploymentImage, ":") && strings.Contains(deploymentImage, "@") {
+					s := strings.SplitAfter(deploymentImage, ":")
+					if strings.Contains(s[1], "@") {
+						t := strings.Split(s[1], "@")
+						version = t[0]
+					}
 				}
 			}
 		}
 	}
 
 	return version
+
+}
+
+// Get triggers version
+func GetTriggersVersion(r Resource, isOpenShift bool) string {
+	version := ""
+	namespacesToCheck := []string{"tekton-pipelines"}
+
+	if isOpenShift {
+		namespacesToCheck = append(namespacesToCheck, "openshift-pipelines")
+	}
+
+	// Go through possible namespaces Tekton Triggers is installed in
+	// For each namespace if the version is empty then it goes through to get the deployments within that namespace to try and locate the Tekton Triggers deployment to get the version
+	for _, versionAttempt := range namespacesToCheck {
+		if version == "" {
+			version = getDeployments(r, "triggers", versionAttempt)
+		}
+	}
+
+	if version == "" {
+		logging.Log.Error("Error getting the Tekton Trigggers deployment version. Version is unknown")
+		return ""
+	}
+	return version
+}
+
+// Check whether Tekton Triggers is installed
+func IsTriggersInstalled(r Resource, isOpenShift bool) bool {
```

Comment:
Is this needed? Can we not try to get the version, and if not found assume it's not installed?

Reply:
Yes it is needed
If the version is not found that could be because of multiple reasons including errors in the dashboard as well as Triggers changing their versioning code 
Could also indicate that a user is using a very old version of triggers if an error with version of triggers is displayed